shell/mixins/resource-manager.js

```diff
@@ -45,24 +48,40 @@ export default {
         // this is the flag/variable that we need to apply to all places that rely on this data. Ex: LabeledSelect
         this.isLoadingSecondaryResources = true;
         const hash = await allHashSettled(requests);
+        const types = Object.keys(hash);
 
-        Object.keys(hash).forEach((type) => {
+        for (let i = 0; i < types.length; i++) {
+          const type = types[i];
           const status = hash[type].status;
           // if it's namespaced, we get the data on 'items' prop, for non-namespaced it's  'data' prop...
           const requestData = hash[type].value.items || hash[type].value.data || hash[type].value;
 
           if (status === 'fulfilled' && resourceData.data[type] && resourceData.data[type].applyTo?.length) {
-            resourceData.data[type].applyTo.forEach((apply) => {
+            for (let y = 0; y < resourceData.data[type].applyTo.length; y++) {
+              const apply = resourceData.data[type].applyTo[y];
+              let resources = requestData;
+
+              if (apply.classify) {
+                // we need to add the type as a obj prop if it's not included... On PVC's in wasn't...
+                if (!requestData[0]?.type) {
+                  requestData.forEach((item, index) => {
+                    requestData[index].type = type;
```

Comment:
I've got to the bottom of why this is needed (and the `detailLocation` in PVC model below).

The response to `k8s/clusters/c-m-v2fqwvpt/v1/persistentvolumeclaims` contains different resource than namespaced `/k8s/clusters/c-m-v2fqwvpt/api/v1/namespaces/workspace/persistentvolumeclaims`. The later omits, amongst others, id, links and type which are all things we depend upon.

This might be a blocker, will talk with Neil. Worst case scenario we need to check how the secondary resources that are namespaced are used and don't depend on id

Reply:
I''ve made this more generic and commented to reflect what's happening, it should get us through this PR https://github.com/rancher/dashboard/pull/6860/commits/9620ee54b4b9eb49ec54879fb654d1c4110c551c